// @redux-devtools/dock-monitor/lib/cjs/actions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TOGGLE_VISIBILITY = exports.CHANGE_SIZE = exports.CHANGE_POSITION = exports.CHANGE_MONITOR = void 0;
exports.changeMonitor = changeMonitor;
exports.changePosition = changePosition;
exports.changeSize = changeSize;
exports.toggleVisibility = toggleVisibility;
const TOGGLE_VISIBILITY = exports.TOGGLE_VISIBILITY = '@@redux-devtools-log-monitor/TOGGLE_VISIBILITY';
function toggleVisibility() {
  return {
    type: TOGGLE_VISIBILITY
  };
}
const CHANGE_POSITION = exports.CHANGE_POSITION = '@@redux-devtools-log-monitor/CHANGE_POSITION';
function changePosition() {
  return {
    type: CHANGE_POSITION
  };
}
const CHANGE_SIZE = exports.CHANGE_SIZE = '@@redux-devtools-log-monitor/CHANGE_SIZE';
function changeSize(size) {
  return {
    type: CHANGE_SIZE,
    size: size
  };
}
const CHANGE_MONITOR = exports.CHANGE_MONITOR = '@@redux-devtools-log-monitor/CHANGE_MONITOR';
function changeMonitor() {
  return {
    type: CHANGE_MONITOR
  };
}
